dev/js/actions/commons/login.js

```diff
@@ -24,20 +25,174 @@ export const userLogin = (data, dispatch, cb) => {
             loading: false,
           },
         });
+        STORAGE.setAuthToken(response.message.token);
+        const message = 'Successfully Logged In';
+        dispatch({
+          type: USER_LOGIN_SUCCESS,
+          payload: {
+            login_message: message,
+          },
+        });
+        if (cb) cb(null, response);
+      } else {
+        let message = 'Error Logging In frontend';
+        dispatch({
+          type: USER_LOGIN_FAIL,
+          payload: {
+            error_message: message,
+          },
+        });
+      }
+    })
+    .catch(function (error) {
+      let message = 'Error Logging In backend';
+      dispatch({
+        type: USER_LOGIN_FAIL,
+        payload: {
+          error_message: message,
+        },
+      });
+      if (cb) cb(message, null);
+    });
+};
+
+export const googleLogin = (data, dispatch, cb) => {
+  dispatch({
+    type: SEND_USER_LOGIN_CREDENTIALS,
+    payload: {
+      loading: true,
+    },
+  });
+  API_POST(API_BASE_URL + '/v1/user/google-login', {
+    access_token: data.accessToken,
+  })
+    .then(function (response) {
+      if (response && response.status == 1) {
+        dispatch({
+          type: SEND_USER_LOGIN_CREDENTIALS,
+          payload: {
+            loading: false,
+          },
+        });
+        STORAGE.setAuthToken(response.message.token);
+        const message = 'Successfully Logged In';
+        dispatch({
+          type: USER_LOGIN_SUCCESS,
+          payload: {
+            login_message: message,
+          },
+        });
+        if (cb) cb(null, response);
+      } else {
+        let message = 'Error Logging In frontend';
+        dispatch({
+          type: USER_LOGIN_FAIL,
+          payload: {
+            error_message: message,
+          },
+        });
+      }
+    })
+    .catch(function (error) {
+      let message = 'Error Logging In backend';
+      dispatch({
```

Comment:
entire body of these login API's looks repeated across..
can we make these common for all (g/fb/lnkedin/csutom signup)